compiler/rustc_error_codes/src/error_codes/E0772.md

```diff
@@ -0,0 +1,87 @@
+A pointer to a trait object has some specific lifetime `'1`, but the trait
+object was used in a way that requires the pointer to have a `'static` lifetime.
+
+Example of erroneous code:
+
+```compile_fail,E0772
+trait Truthy {}
+trait Person {}
+
+impl Truthy for bool {}
+
+impl dyn Person {
+    fn is_cool(&self) -> bool {
+        // hey you, you're pretty cool
+        true
+    }
+}
+
+fn get_is_cool<'p>(person: &'p dyn Person) -> impl Truthy {
+    // error: `person` has an anonymous lifetime `'p` but calling
+    //        `print_cool_fn` introduces an implicit `'static` lifetime
+    //        requirement
+    person.is_cool()
+}
+```
+
+The trait object `person` in the function `get_is_cool`, while already being
+behind a pointer with lifetime `'p`, also has it's own lifetime, `'2`.
+
+Lifetime `2` represents the data the trait object may hold inside, for example:
+
+```
+trait MyTrait {}
+
+struct MyStruct<'a>(&'a i32);
+
+impl<'a> MyTrait for MyStruct<'a> {}
+```
+
+In this scenario, if a trait object of `dyn MyTrait + '2` was made from
+`MyStruct<'a>`, the `where` clause `'a: '2` must be upheld. `'a: '2` meaning
+that `'a` must live as long if not longer than `'2`, so that the trait object's
+internal data may be accessed safely.
+
+In the implementation for `dyn Person`, the `'2` lifetime representing the
+internal data was ommitted, meaning that the compiler inferred the lifetime
+`'static`. As a result, the implementation's `is_cool` is inferred by the
+compiler to look like this:
+
+```
+# trait Person {}
+#
```

Comment:
All of these `#`s will cause the terminal output to look strange since (unlike rustdoc) it doesn't currently strip them.

Reply:
No, it works as expected. Try with E0750 for example. ;)

Reply:
Alright, I stand corrected :)